api/autopilot.go

```diff
@@ -4,38 +4,35 @@ import (
 	"errors"
 	"fmt"
 
-	"go.sia.tech/core/types"
 	"go.sia.tech/renterd/internal/utils"
 )
 
 const (
 	// BlocksPerDay defines the amount of blocks that are mined in a day (one
 	// block every 10 minutes roughly)
 	BlocksPerDay = 144
-
-	// DefaultAutopilotID is the id of the autopilot.
-	DefaultAutopilotID = "autopilot"
 )
 
 var (
-	// ErrAutopilotNotFound is returned when an autopilot can't be found.
-	ErrAutopilotNotFound = errors.New("couldn't find autopilot")
-
-	// ErrMaxDowntimeHoursTooHigh is returned if the autopilot config is updated
+	// ErrMaxDowntimeHoursTooHigh is returned if the contracts config is updated
 	// with a value that exceeds the maximum of 99 years.
 	ErrMaxDowntimeHoursTooHigh = errors.New("MaxDowntimeHours is too high, exceeds max value of 99 years")
+
+	// ErrInvalidReleaseVersion is returned if the version is an invalid release
+	// string.
+	ErrInvalidReleaseVersion = errors.New("invalid release version")
 )
 
 type (
-	// Autopilot contains the autopilot's config and current period.
+	// Autopilot contains its configuration as well as the current period.
 	Autopilot struct {
-		ID            string          `json:"id"`
-		Config        AutopilotConfig `json:"config"`
-		CurrentPeriod uint64          `json:"currentPeriod"`
+		CurrentPeriod uint64 `json:"currentPeriod"`
```

Comment:
Ditto: is there a solid benefit to aligning contract periods? Always creating contracts for the desired duration doesn't seem too bad. It might help with repair churn since the renter won't need to refresh all of their contracts at the exact same time.

Reply:
The main reason was budgeting. Since you can't limit period spending without periods. But we got rid of that too by removing the allowance so we might as well embrace the chaos.
Tracking itself (e.g. you locked away X funds within the last 4 weeks and unlocked Y) should still be doable indeed.